catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java

```diff
@@ -153,6 +147,13 @@ public DynamicSchemaResolver() {
         fieldsCache.add(Metacard.ID + SchemaFields.TEXT_SUFFIX + SchemaFields.TOKENIZED
                 + SchemaFields.HAS_CASE);
         fieldsCache.add(Metacard.TAGS + SchemaFields.TEXT_SUFFIX);
+
+        anyTextFieldsCache.add(Metacard.METADATA + SchemaFields.TEXT_SUFFIX);
+        BasicTypes.BASIC_METACARD.getAttributeDescriptors()
+                .stream()
+                .filter(descriptor -> BasicTypes.STRING_TYPE.equals(descriptor.getType()))
+                .forEach(stringDescriptor -> anyTextFieldsCache.add(
```

Comment:
Instead of a `forEach` with a lambda to add to the set, you could do a `map` followed by a `collect(Collectors.toSet())` and `addAll` to `anyTextFieldsCache`.